src/main/java/brandkon/brand/BrandService.java

```diff
@@ -0,0 +1,53 @@
+package brandkon.brand;
+
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class BrandService {
+
+    private final BrandRepository brandRepository;
+    private final BrandCategoryRepository brandCategoryRepository;
+
+    public BrandService(BrandRepository brandRepository, BrandCategoryRepository brandCategoryRepository) {
+        this.brandRepository = brandRepository;
+        this.brandCategoryRepository = brandCategoryRepository;
+    }
+
+    // N:M관계일때
+    public List<BrandResponse> read(String category) {
+        List<BrandCategory> byCategorySlug = brandCategoryRepository.findByCategorySlug(category);
+        return byCategorySlug.stream().map(brandCategory ->
+                new BrandResponse(
+                        brandCategory.getBrand().getId(),
+                        brandCategory.getBrand().getName(),
+                        brandCategory.getBrand().getImageUrl()
+                )).toList();
+    }
+
+    /*
+    * 1:N 관계일때
+    public List<BrandResponse> read(String category) {
+        List<Brand> categorySlug = brandRepository.findByCategorySlug(category);
+        return categorySlug.stream().map(brand -> new BrandResponse(
+                brand.getId(),
+                brand.getName(),
+                brand.getImageUrl()
+        )).toList();
+    }*/
+
+    /* 1:n 관계
+    public BrandResponse oneBrandRead(Long brandId) {
+        Brand brand = brandRepository.findById(brandId).orElseThrow();
+        return new BrandResponse(brandId, brand.getName(), brand.getImageUrl());
+    }*/
+
+    public BrandResponse oneBrandRead(Long brandId) {
+        BrandCategory brandCategory = brandCategoryRepository.findById(brandId).orElseThrow();
```

Comment:
하나의 브랜드를 brandId로 찾는 거라면 BrandRepository의 findById를 사용하는 것이 더 적절할 것 같아요.
그런데 API 설계상 하나의 브랜드만 찾을 일이 없을 것 같아요.

추가로, 에러 메시지는 상세하게 알려 주는 것이 더 좋아요
```suggestion
        BrandCategory brandCategory = brandCategoryRepository.findById(brandId).orElseThrow(() -> new NoSuchElementException("존재하지 않는 브랜드 id: " + brandId));
```